### ISSUE TITLE
Switch from markdownlint-cli to markdownlint-cli2

### ISSUE THREAD

**Issue by koppor** [avto-dev/markdown-lint#17]
### Is your feature request related to a problem?

`markdown-lint` supports configuration files per directory.

Example:

root directory

```yaml
default: true
MD013: false
```

sub directory `docs/decisions`

```yaml
default: true
MD013: false
MD024: false
```

When running on such a repository and a file in `docs/decisions` violates MD024, [markdown-cli](https://github.com/igorshubovych/markdownlint-cli) reports an error:

```bash
$ npx markdownlint-cli '**/*.md' --ignore node_modules
docs/decisions/0010-support-categories.md:103 MD024/no-duplicate-heading/no-duplicate-header Multiple headings with the same content [Context: "#### Examples"]
docs/decisions/0015-include-consulting-informed-of-raci.md:32 MD024/no-duplicate-heading/no-duplicate-header Multiple headings with the same content [Context: "### Include "Consulted" and "I..."]
```

Since this action is based on markdownlint-cli, it does not support multiple markdownlint configuration files - and thus also reports errors.

When running `markdown-cli2`, no error is reported:

```bash
$ npx markdownlint-cli2 "**/*.md" "#node_modules"
markdownlint-cli2 v0.4.0 (markdownlint v0.25.1)
Finding: **/*.md !node_modules
Linting: 31 file(s)
Summary: 0 error(s)
```

### Describe the solution you'd like

Change from markdownlint-cli to [markdown-cli2](https://github.com/DavidAnson/markdownlint-cli2).

## Additional context

Blog post on markdownlint-cli2: https://dlaa.me/blog/post/markdownlintcli2

**Comment by koppor** [comment]
I drafted a PR at https://github.com/koppor/markdown-lint/pull/1. Will create a PR to this repository as soon as upstream issues are fixed.

**Comment by eldario** [comment]
@koppor 

I apologize for the **long answer**.

Did I understand you correctly.
In this issue, you suggest switching to the official image `davidanson/markdownlint-cli2`

Now the check will go through the configuration file inside the checked directory.

For example:
```
src/
├─ docs/
│  └─ api/
│     ├─ changelog.yaml
│     ├─ README.md
│     ├─ API.md
│     ├─ UPDATES.md
│     └─ version2/
│        ├─ changelog.yaml
│        ├─ VERSION_2.md
│        └─ TO_UPDATE/
│           └─ UPDATES_V2.md
├─ changelog.yaml
└─ CHANGELOG.md

```

#### Directory `/src`
- `CHANGELOG.md` uses `./src/changelog.yaml`

#### Directory `/src/docs/api`
- `README.md` uses `./src/docs/api/changelog.yaml`
- `API.md` uses `./src/docs/api/changelog.yaml`
- `UPDATES.md` uses `./src/docs/api/changelog.yaml`

#### Directory `/src/docs/api/version2`
- `VERSION_2.md` uses `./src/docs/api/version2/changelog.yaml`

#### Directory `/src/docs/api/version2/TO_UPDATE`
- `UPDATES_V2.md` uses `./src/docs/api/version2/changelog.yaml`

**Comment by koppor** [comment]
@eldario Partially. The configuration file is `.markdownlint.yaml` if not configured differently. Thus, directory `/src/docs/api/version2/TO_UPDATE` tries to read `src/docs/api/version2/TO_UPDATE/.markdownlint.yaml`. If that does not exist, it tries to read `src/docs/api/version2/markdownlint.yaml`.

I still need to work on https://github.com/DavidAnson/markdownlint-cli2/issues/49 and https://github.com/DavidAnson/markdownlint-cli2/issues/50 to be able to get `markdownlint-cli2` running "properly" with this custom CHANGELOG configurations. In my draft-PR, I need to copy configurations files back and forth. E.g., https://github.com/koppor/markdown-lint/blob/c2a97fcebfb4d65d83ffc2be13fb6f6d76675d0a/.github/workflows/ci.yml#L31

**Comment by gomzyakov** [comment]
@koppor @eldario shall we close this issue?

**Comment by koppor** [comment]
You can close this issue. The work-around is to use this action for validating CHANGELOG.md files and [markdownlint-cli2-action](https://github.com/DavidAnson/markdownlint-cli2-action#markdownlint-cli2-action) for validating all other Markdown files.
